### PR TITLE
Azure `Page` type `nextLink` properties should be `readOnly: true`

### PR DESCRIPTION
**Pull request by daviwil**
This PR fixes https://github.com/Azure/typespec-azure-pr/issues/3924 by marking `nextLink` properties with `@visibility("read")` so that the OpenAPI output for these properties contains `readOnly: true`.

### CODE REVIEW

File: .changeset/next-link-read-only-2024-1-7-14-13-43.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@azure-tools/typespec-azure-core": patch
```
Comment (timotheeguerin) [inline .changeset/next-link-read-only-2024-1-7-14-13-43.md:2]:
```suggestion
"@azure-tools/typespec-azure-core": minor
```

Comment (timotheeguerin) [reply .changeset/next-link-read-only-2024-1-7-14-13-43.md:2]: As I think this is not really a fix but a change

**Review by timotheeguerin — APPROVED** [review]
Looks good, is there any breaking change issue with this? (apart from the swagger spec will need to be regenerated in spec repos)

File: .changeset/next-link-read-only-2024-1-7-14-13-43.md (hunk 1 above)
Comment (daviwil) [reply .changeset/next-link-read-only-2024-1-7-14-13-43.md:2]: Good point, thanks!

**Comment by daviwil** [comment]
@timotheeguerin We probably should mention it.  This is something that should have been set to begin with but got missed during Azure Core development. @allenjzhang and I discussed that it's worth introducing a break.  Anyone else have opinions here?  @markcowl @johanste

**Review by markcowl — APPROVED** [review]

**Pull request merged**